api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/request/MixedMetricsRequest.java

```diff
@@ -42,15 +42,27 @@
 
 
     public List<Gauge> getGaugeMetrics() {
-        return ImmutableList.copyOf(gaugeMetrics);
+        if(gaugeMetrics != null){
+            return ImmutableList.copyOf(gaugeMetrics);
+        }
+
+        return null;
     }
 
     public List<Availability> getAvailabilityMetrics() {
-        return ImmutableList.copyOf(availabilityMetrics);
+        if (availabilityMetrics != null) {
+            return ImmutableList.copyOf(availabilityMetrics);
+        }
+
+        return null;
     }
 
     public List<Counter> getCounters() {
-        return ImmutableList.copyOf(counters);
+        if (counters != null) {
+            return ImmutableList.copyOf(counters);
+        }
+
+        return null;
     }
 
```

Comment:
I dropped the immutable portion out of the request object and closer to the metrics service calls. That should help checking for null and empty content without incurring the copy penalty.